h2o-algos/src/main/java/hex/generic/PojoLoader.java

```diff
@@ -18,23 +18,35 @@ class PojoLoader {
 
     private static final String POJO_EXT = ".java";
 
-    static GenModel loadPojoFromSourceCode(ByteVec sourceVec, Key<Frame> pojoKey) throws IOException {
-        String pojoCode = IOUtils.toString(sourceVec.openStream());
+    static GenModel loadPojoFromSourceCode(ByteVec sourceVec, Key<Frame> pojoKey, String modelId) throws IOException {
+        final String pojoCode = IOUtils.toString(sourceVec.openStream());
+        final String className;
         try {
-            String path = URI.create(pojoKey.toString()).getPath();
-            String fileName = new File(path).getName();
-            if (fileName.endsWith(POJO_EXT)) {
-                fileName = fileName.substring(0, fileName.length() - POJO_EXT.length());
-            }
-            return compileAndInstantiate(fileName, pojoCode);
+            className = inferClassName(pojoKey, modelId);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to automatically infer POJO class name, " +
+                    "please specify the POJO class name explicitly be setting `model_id` parameter.");
```

Comment:
At first sight, this exception message confuses me. It seems to me, that if there happens an exception during inferClassName() commands, it doesn't matter whether modelId is set to something or whether it is a null.

Reply:
you are right, this doesn't do anything!

Reply:
fixed, thank you for pointing it out